From c0b77078c3ef25457f366cc0c27e9600f92a5a9c Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 29 Jun 2022 11:36:37 +0300
Subject: [PATCH 004/102] arm: s32cc: Set SERDES_INVALID on 0 instead of -1

This is needed to avoid false positives when the validity
of the SerDes mode is checked using mode & PCI_EP or
mode & PCIE_RC. SERDES_INVALID will mark the mode as Root Complex
and Endpoint at the same time.

Issue: ALB-8357
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/include/s32-cc/serdes_hwconfig.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-s32/include/s32-cc/serdes_hwconfig.h b/arch/arm/mach-s32/include/s32-cc/serdes_hwconfig.h
index a7c7cd2b8a..e4ea66450b 100644
--- a/arch/arm/mach-s32/include/s32-cc/serdes_hwconfig.h
+++ b/arch/arm/mach-s32/include/s32-cc/serdes_hwconfig.h
@@ -28,7 +28,7 @@ enum serdes_phy_mode {
 };
 
 enum serdes_dev_type {
-	SERDES_INVALID = -1,
+	SERDES_INVALID = 0,
 	PCIE_EP = 0x1, /* EP mode is 0x0, use 0x1 to allow us to use masks */
 	PCIE_RC = 0x4,
 	SGMII = 0x10, /* outside range of PE0_GEN_CTRL_1:DEVICE_TYPE */
-- 
2.17.1

